### DISCUSSION TITLE
"SQLITE_ERROR: no such table" in testing

### DISCUSSION THREAD

**Discussion by backspacerhino** [adonisjs/core#1527] [Help] [answered]
Hi, I am following [THIS](https://preview.adonisjs.com/blog/running-tests-in-adonisjs-v5/) blog post on running japa tests in AdonisV5

I've installed and setup my sqlite3 to work in ":memory:".

When I run `node build/japaFile.js` my migrations run successfully however when I try to read/write to any of my tables I get an error  `SQLITE_ERROR: no such table: accounts`

Edit:
Also, it appears as if japa doesn't care about .env.testing file

Edit2:
I think it might have to do with `:memory:` database being wiped after connection closes and Adonis might not keep the connection alive after migrations are done. In AdonisV4.1 I had to specify `keepAlive:true` which is not available here

**Comment by backspacerhino** [comment, marked as answer]
Any solutions to this?

**Reply by thetutlage** [reply]
> Also, it appears as if japa doesn't care about .env.testing file

Yes, coz Japa is a framework agnostic test runner and hence doesn't deal with loading env files. You can run your tests as `NODE_ENV=testing node build/japaFile.js` and then AdonisJS will make sure to read the `.env.testing` file for you.

>  I think it might have to do with :memory: database being wiped after connection closes and Adonis might not keep the connection alive after migrations are done. In AdonisV4.1 I had to specify keepAlive:true which is not available here

Yes, that is the reason. I don't think there is any way to make migrations not close the connection (as of today). The only option is not to use `:memory:` database.

**Reply by backspacerhino** [reply]
I see, thank you for the answer.

I will use another approach instead of `:memory:` for the time being
